### DISCUSSION TITLE
Generic chat UI like Chainlit or Streamlit for C#?

### DISCUSSION THREAD

**Discussion by benceblaske** [microsoft/semantic-kernel#6401] [Q&A] [answered]
I have some experience building chat based experiences using generic UI frameworks like Streamlit or Chainlit (there are many others), but these are available only for Python. Since the C# SDK of Semantic Kernel is the most mature, I am looking for something like the above Python frameworks. Any suggestions for such, that can be used with SK? Thanks!

**Comment by dmytrostruk** [comment, marked as answer]
@benceblaske The closest thing for C# to build chat based UI together with Semantic Kernel would be Blazor. You can use it together with HTML/CSS/JS, which means that in theory you can combine it with other available JS UI frameworks. 

> Since the C# SDK of Semantic Kernel is the most mature

We just released Python v1 version and at this point it should be up to date with all C# features. You can try to handle your scenario with Python version together with Streamlit or Chainlit and share your experience with us, it would be really interesting to learn about it!

**Reply by benceblaske** [reply]
Thanks! Yes, I was looking into Blazor, but I was hoping for a more polished, ready made and proven solutions like Chainlit or Streamlit. :) I would have assumed that by now at least something would exist for C#, but I was searching extensively on Github and haven't found anything yet... 

Indeed, my goal would to transition back to Python, as it's much easier for me to code in Python, and also many more "auxiliary" things exis on the Python side of things. I was trying to migrate my sample POC to Python (https://github.com/bbence84/semantic_kernel_copilot_demo), but there are still things missing: especially the Handlebars planner, and I have started to play around with Agents in SK (I have prior Autogen experience, and I think that I could leverage agent collaboration for many interesting things). Agents also do not exist yet on the Python side. But yes, I hope that eventually I can return to Python, thanks for the nudge! :)

**Comment by stephentoub** [comment]
cc: @SteveSandersonMS

**Reply by SteveSandersonMS** [reply]
Moved this to the ASP.NET/Blazor issue tracker: https://github.com/dotnet/aspnetcore/issues/56265
